### PR TITLE
Add warmup style: cosine.

### PR DESCRIPTION
**Pull request by lavinal712**
This PR introduces a new warmup style for learning rate scheduling: cosine warmup. Users can now specify `warmup_style: "cosine"` in the optimizer configuration.

Example script

```bash
#!/bin/bash

set -x

MODEL_PATH=Qwen/Qwen2.5-VL-3B-Instruct  # replace it with your local file path

python3 -m verl.trainer.main \
    config=examples/config.yaml \
    data.train_files=hiyouga/geometry3k@train \
    data.val_files=hiyouga/geometry3k@test \
    worker.actor.model.model_path=${MODEL_PATH} \
    worker.actor.optim.warmup_style=cosine \
    worker.actor.optim.training_steps=60 \
    trainer.experiment_name=qwen2_5_vl_3b_geo_grpo \
    trainer.n_gpus_per_node=4
```

update:

```bash
#!/bin/bash

set -x

MODEL_PATH=Qwen/Qwen2.5-VL-3B-Instruct  # replace it with your local file path

python3 -m verl.trainer.main \
    config=examples/config.yaml \
    data.train_files=hiyouga/geometry3k@train \
    data.val_files=hiyouga/geometry3k@test \
    worker.actor.model.model_path=${MODEL_PATH} \
    worker.actor.optim.lr_scheduler_type=cosine \
    worker.actor.optim.training_steps=60 \
    trainer.experiment_name=qwen2_5_vl_3b_geo_grpo \
    trainer.n_gpus_per_node=4
```

### CODE REVIEW

File: verl/workers/fsdp_workers.py [hunk 1]
```diff
@@ -311,9 +311,23 @@ def _build_model_optimizer(
...
                 num_warmup_steps = int(optim_config.lr_warmup_ratio * optim_config.training_steps)
 
-            self.lr_scheduler = get_constant_schedule_with_warmup(
-                optimizer=self.optimizer, num_warmup_steps=num_warmup_steps
-            )
+            if optim_config.warmup_style == "constant":
+                self.lr_scheduler = get_constant_schedule_with_warmup(
+                    optimizer=self.optimizer, num_warmup_steps=num_warmup_steps
+                )
+            elif optim_config.warmup_style == "cosine":
```
Comment (hiyouga) [inline verl/workers/fsdp_workers.py]: It should be `scheduler_type` instead: https://github.com/volcengine/verl/blob/main/verl/trainer/config/optim/fsdp.yaml#L39

File: verl/workers/actor/config.py [hunk 2]
```diff
@@ -50,6 +50,7 @@ class OptimConfig:
     lr_warmup_steps: Optional[int] = None
     min_lr_ratio: Optional[float] = None
     warmup_style: str = "constant"
+    num_cycles: float = 0.5
```
Comment (hiyouga) [inline verl/workers/actor/config.py]: It is not a common setting, we can remove it and set the num cycles to 0.5 by default

**Review by hiyouga — APPROVED** [review]
